misc/ballerinalang-data-mapper/src/main/java/org/ballerinalang/datamapper/AIDataMapperCodeActionUtil.java

```diff
@@ -274,61 +303,210 @@ public static Symbol findSymbol(List<TypeSymbol> fileContentSymbols) {
      * @param context
      * @param foundTypeLeft  {@link String}
      * @param foundTypeRight {@link String}
+     * @param syntaxTree
+     * @param semanticModel
      * @return function string with mapped schemas
      * @throws IOException throws if error occurred when getting mapped function
      */
     private static String getGeneratedRecordMapping(CodeActionContext context, String foundTypeLeft,
                                                     String foundTypeRight,
-                                                    Symbol lftTypeSymbol, Symbol rhsTypeSymbol)
+                                                    Symbol lftTypeSymbol, Symbol rhsTypeSymbol, SyntaxTree syntaxTree, SemanticModel semanticModel)
 
             throws IOException {
         JsonObject rightRecordJSON = new JsonObject();
         JsonObject leftRecordJSON = new JsonObject();
 
-        // Schema 1
-        TypeSymbol rhsSymbol = ((TypeReferenceTypeSymbol) rhsTypeSymbol).typeDescriptor();
-        if ("RECORD".equals(rhsSymbol.typeKind().name())) {
-            RecordTypeSymbol rightSymbol = (RecordTypeSymbol) rhsSymbol;
-            Map<String, RecordFieldSymbol> rightSchemaFields = rightSymbol.fieldDescriptors();
+        List<RecordTypeSymbol> symbolList = checkMappingCapability(lftTypeSymbol, rhsTypeSymbol);
+        if (!symbolList.contains(null)) {
```

Comment:
Shall we avoid adding null entries to the list? Is there any specific reason/ limitation for adding the nulls

Reply:
Fixed this. Stop adding null entries.